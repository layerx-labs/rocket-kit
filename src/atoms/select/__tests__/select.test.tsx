import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { Select } from '../../..';

describe('Select Default', () => {
  const options = [
    { value: 'option_1', name: 'Option 1' },
    { value: 'option_2', name: 'Option 2' },
    { value: 'option_3', name: 'Option 3' },
  ];

  it('renders', () => {
    const { asFragment } = render(<Select name="title" options={options} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
