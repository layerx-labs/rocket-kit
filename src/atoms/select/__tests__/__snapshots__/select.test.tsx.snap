// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Default renders 1`] = `
<DocumentFragment>
  .c0 {
  border: 1px solid hsl(0,0%,48%);
  border-radius: 6px;
  background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' width='32' height='32' viewBox='0 0 32 32'><path fill='hsl(0, 0%, 85%)' d='M10.827 12.387l5.173 5.173 5.173-5.173c0.52-0.52 1.36-0.52 1.88 0v0c0.52 0.52 0.52 1.36 0 1.88l-6.12 6.12c-0.52 0.52-1.36 0.52-1.88 0l-6.12-6.12c-0.52-0.52-0.52-1.36 0-1.88v0c0.52-0.507 1.373-0.52 1.893 0z'></path></svg>");
  background-position: calc(100% - 15px) center;
  background-repeat: no-repeat;
  background-size: 1.875rem;
  background-color: transparent;
  height: 3.125rem;
  padding: 0 4.0625rem 0 0.9375rem;
  font-size: 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0:hover {
  border-color: hsl(256,55%,43%);
}

.c0:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 hsl(0,0%,16%);
}

.c0:disabled {
  background-color: #ededed;
  color: hsl(0,0%,48%);
  pointer-events: none;
}

<select
    class="c0"
    name="title"
  >
    <option
      value="option_1"
    >
      Option 1
    </option>
    <option
      value="option_2"
    >
      Option 2
    </option>
    <option
      value="option_3"
    >
      Option 3
    </option>
  </select>
</DocumentFragment>
`;
